addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -350,7 +352,21 @@ private int isRedirected(String payload, HttpMessage msg) {
         // Location: http://www.example.org/index.php
         String value = msg.getResponseHeader().getHeader(HttpHeader.LOCATION);
         if (checkPayload(value, payload)) {
-            return REDIRECT_LOCATION_HEADER;
+            URI locUri = null;
+            URI payloadUri = null;
+            try {
+                locUri = new URI(value.replace("%2e", "."));
```

Comment:
I think if we use `org.apache.commons.httpclient.URI` then we don't need to do the replace.